library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt

```diff
@@ -0,0 +1,400 @@
+ * - Set calendar header color: headerColor="@color/[color]"
+ * - Set calendar header label color: headerLabelColor="@color/[color]"
+ * - Set previous button resource: previousButtonSrc="@drawable/[drawable]"
+ * - Ser forward button resource: forwardButtonSrc="@drawable/[drawable]"
+ * - Set today label color: todayLabelColor="@color/[color]"
+ * - Set selection color: selectionColor="@color/[color]"
+ *
+ * Created by Mateusz Kornakiewicz on 23.05.2017.
+ */
+
+class CalendarView @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0,
+        properties: CalendarProperties? = null
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    private var currentPage: Int = 0
+
+    private var calendarProperties: CalendarProperties = CalendarProperties(context)
+    private var calendarPageAdapter: CalendarPageAdapter = CalendarPageAdapter(context, calendarProperties)
+
+    private val onPageChangeListener = object : ViewPager.OnPageChangeListener {
+        /**
+         * This method set calendar header label
+         *
+         * @param position Current ViewPager position
+         * @see ViewPager.OnPageChangeListener
+         */
+        override fun onPageSelected(position: Int) {
+            val calendar = calendarProperties.firstPageCalendarDate?.clone() as Calendar
+            calendar.add(Calendar.MONTH, position)
+
+            if (!isScrollingLimited(calendar, position)) {
+                setHeaderName(calendar, position)
+            }
+        }
+
+        override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
+        override fun onPageScrollStateChanged(state: Int) {}
+    }
+
+    /**
+     * @return List of Calendar object representing a selected dates
+     */
+    var selectedDates: List<Calendar>
+        get() = calendarPageAdapter.selectedDays
+                .mapNotNull { it.calendar }
+        set(selectedDates) {
+            calendarProperties.selectDays(selectedDates)
+            calendarPageAdapter.notifyDataSetChanged()
+        }
+
+    /**
+     * @return Calendar object representing a selected date
+     */
+    val selectedDate: Calendar?
+        get() = calendarPageAdapter.selectedDays.map { it.calendar }.first()
+
+    /**
+     * @return Calendar object representing a date of current calendar page
+     */
+    private val currentPageDate: Calendar
+        get() {
+            val calendar = calendarProperties.firstPageCalendarDate?.clone() as Calendar
+            return calendar.apply {
+                set(Calendar.DAY_OF_MONTH, 1)
+                add(Calendar.MONTH, calendarViewPager.currentItem)
+            }
+        }
+
+    init {
+        properties?.let { this.calendarProperties = properties }
+        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
```

Comment:
Zadziała tutaj LayoutInflater.from(context).inflate(R.layout.calendar_view, this, false) zamiast tych 2 linijek?